packages/commonwealth/client/scripts/views/modals/edit_topic_modal.tsx

```diff
@@ -51,9 +51,18 @@ export const EditTopicModal = ({
     featuredInSidebarProp,
   );
   const [name, setName] = useState<string>(nameProp);
+  const [image, setImage] = useState<string | null>(null);
 
   const { isAddedToHomeScreen } = useAppStatus();
 
+  const markdownImageRegex = useMemo(() => /!\[image\]\((.*?)\)/, []);
+
+  //checks if there is an image in the description to show/delete separately in the modal
+  useMemo(() => {
```

Comment:
I also think we don't need this useMemo, I am pretty sure we can move it into where we set the description so:

```typescript
<CWTextInput
          label="Description"
          name="description"
          tabIndex={2}
          value={description}
          onInput={(e) => {
            setDescription(e.target.value);
          }}
        />
```

becomes:
```typescript
<CWTextInput
          label="Description"
          name="description"
          tabIndex={2}
          value={description}
          onInput={(e) => {
            setDescription(e.target.value);
            const match = description.match(markdownImageRegex);
            setImage(match ? match[1] : null);
          }}
        />
```

Reply:
@kurtisassad Yeah, you're right. Thanks for catching those.